tests/unit/fetch-test.js

```diff
@@ -0,0 +1,25 @@
+var simple = require('simple-mock')
+var test = require('tape')
+
+var fetch = require('../../lib/fetch')
+var internals = fetch.internals
+
+test('fetch', function (t) {
+  simple.mock(internals, 'fetchProperties').resolveWith({})
+
+  fetch({
+    baseUrl: 'http://example.com',
+    session: {
+      id: 'abc4567'
+    }
+  }, 'path')
+
+  t.deepEqual(internals.fetchProperties.lastCall.arg, {
+    url: 'http://example.com/session/account',
+    bearerToken: 'abc4567',
+    path: 'path'
+  }, 'calls fetchProperties with account url')
```

Comment:
that we only check if the other "units" have been called correctly, and stub their behavior, so we really only test the `.fetch()` method here, and none of the other methods called internally. 

Sometimes devs tend to overdo it :) And maybe I did here, but I wanted to show what I how it should be :tm: